hw2/05-form.js

```diff
@@ -0,0 +1,59 @@
+function validateForm() {
+  const formatName = /[`!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~0123456789]/;
+  const formatEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  let fieldName = document.forms["form1"]["Username"].value;
+  let fieldEmail = document.forms["form1"]["email"].value;
+
+  let validBool = true;
+  if (fieldName === "" || fieldName.match(formatName)) {
+    window.alert("Please enter a valid Name");
```

Comment:
You can use the `required` attribute in the HTML to avoid handling cases like this in the JavaScript.